File: Moviles/WorkSpace Moviles/Teoria/TeorioMenuContextual/app/src/main/java/com/example/menuscontxt/MainActivity.java
package com.example.menuscontxt;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        /*
        if ((item.getItemId()) == R.id.menuGrupo1){
            Log.d("depuracdion", "Opcion 1 seleccionada");
        }
 º      */
        Snackbar.make(findViewById(R.id.lyPrincipal), item.getTitle().toString() , Snackbar.LENGTH_LONG).show();
        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu,menu);
        return true;
    }

    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {
        return onOptionsItemSelected(item);
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
        setContentView(R.layout.activity_main);

        registerForContextMenu((findViewById(R.id.lyPrincipal)));
    }
}

File: Moviles/WorkSpace Moviles/Ejercicio10menu_confirmacion/app/src/main/java/com/example/ejercicio10menus_confirmacion/MainActivity.kt
package com.example.ejercicio10menus_confirmacion

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AlertDialog
import com.example.ejercicio10menus_confirmacion.databinding.ActivityMainBinding
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    lateinit var binding:ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        registerForContextMenu(binding.lLayout)

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        confirmar(item.title.toString())
        return true
    }

    private fun confirmar(color:String) {
        var colorSeleccionado: Int =0
        when(color){
            "Rojo" -> colorSeleccionado=Color.RED;
            "Verde" -> colorSeleccionado=Color.GREEN;
            "Amarillo" -> colorSeleccionado=Color.YELLOW;
            "Blanco" -> colorSeleccionado=Color.WHITE;
        }
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Cambio de color")
        builder.setMessage("Vas a cambiar el color de fondo a "+color+", estas seguro??")
        builder.setPositiveButton("Si"){dialog, which ->
            binding.lLayout.setBackgroundColor(colorSeleccionado)
            dialog.cancel()
        }
        builder.setNegativeButton("no"){dialog, which->
            Snackbar.make(binding.root, "Pues vale", Snackbar.LENGTH_SHORT).show()
            dialog.cancel()
        }
        builder.show()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.mi_menu, menu)
        return true
    }

    override fun onCreateContextMenu(
        menu: ContextMenu?,
        v: View?,
        menuInfo: ContextMenu.ContextMenuInfo?
    ) {
        menuInflater.inflate(R.menu.menu_contextual, menu)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        confirmar(item.title.toString())
        return true
    }
}